revolutionpi/encoder.go

```diff
@@ -0,0 +1,101 @@
+//go:build linux
+
+// Package revolutionpi implements the Revolution Pi board GPIO pins.
+package revolutionpi
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"go.viam.com/rdk/components/encoder"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/resource"
+)
+
+type revolutionPiEncoder struct {
+	resource.Named
+	resource.TriviallyReconfigurable
+	pin *digitalInterrupt
+}
+
+// EncoderModel is the model triplet for the rev-pi board encoder.
+var EncoderModel = resource.NewModel("viam-labs", "kunbus", "revolutionpi-encoder")
+
+// EncoderConfig is the config for the rev-pi board encoder.
+type EncoderConfig struct {
+	resource.TriviallyValidateConfig
+	Name string `json:"address_name,omitempty"`
```

Comment:
:grimacing: What happens if you omit the name? Won't that cause problems using it?

...also, what is an "address_name"? I don't know the names of any addresses on a board; maybe put in a comment explaining where to look these up.

Reply:
since the revolution pi has all of the addresses in memory tied to a name I felt like address_name was an okayish way to say that. At first I was just gonna call it name but then it felt weird for a component to have multiple fields called name. pin_name would also work. 

Also wasn't super focused on the name in this pr(mentioned that in another comment) since it technically had not been scoped yet.

Reply:
tbh just changed the name since other parts of the code/readme use "pin name"